proxmoxtf/provider/schema.go

```diff
@@ -98,5 +98,35 @@ func nestedProviderSchema() map[string]*schema.Schema {
 			},
 			ValidateFunc: validation.StringIsNotEmpty,
 		},
+		mkProviderSSHUsername: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Used to override the ssh username",
+			DefaultFunc: schema.MultiEnvDefaultFunc(
+				[]string{"PROXMOX_VE_SSHUSERNAME", "PM_VE_SSHUSERNAME"},
+				nil,
+			),
+			AtLeastOneOf: []string{
+				mkProviderSSHUsername,
+				fmt.Sprintf("%s.0.%s", mkProviderVirtualEnvironment, mkProviderSSHUsername),
+			},
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Since the field is optional it should not cause any compatibility issues. If it's not set the old way of getting the ssh user is still used. This is the logic for it:

https://github.com/zoop-btc/terraform-provider-proxmox/blob/8637dba2db97905b50e7cdc2106c7c55e8282321/proxmox/virtual_environment_nodes.go#L198-L202

Reply:
I don't get what the issue would be here. The validation is only checked once a value was provided in the first place as far as I know.

Reply:
Hmmm, I might be wrong, but let me double check with a use case I have in mind. Will keep this conversation open for now.